extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -278,6 +278,14 @@ VertxWebRouterBuildItem initializeRouter(VertxHttpRecorder recorder,
             }
         }
 
+        /**
+         * To create mainrouter when `${quarkus.http.root-path}` is not {@literal /}
+         * Refer https://github.com/quarkusio/quarkus/issues/34261
+         */
+        if (!httpBuildTimeConfig.rootPath.equals("/") && !mainRouterCreated) {
+            mainRouter = recorder.initializeRouter(vertx.getVertx());
```

Comment:
We need to be sure it's used correctly.

Reply:
Is there any specific usage you are concerned about ?

Reply:
The test covers it. Let's see what the CI says.